src/data/nat/with_bot.lean

```diff
@@ -11,43 +11,63 @@ import algebra.order.monoid.with_top
 
 Lemmas about the type of natural numbers with a bottom element adjoined.
 -/
+
 namespace nat
 
-lemma with_bot.add_eq_zero_iff : ∀ {n m : with_bot ℕ}, n + m = 0 ↔ n = 0 ∧ m = 0
-| none     m        := iff_of_false dec_trivial (λ h, absurd h.1 dec_trivial)
-| n        none     := iff_of_false (by cases n; exact dec_trivial)
-  (λ h, absurd h.2 dec_trivial)
-| (some n) (some m) := show (n + m : with_bot ℕ) = (0 : ℕ) ↔ (n : with_bot ℕ) = (0 : ℕ) ∧
-    (m : with_bot ℕ) = (0 : ℕ),
-  by rw [← with_bot.coe_add, with_bot.coe_eq_coe, with_bot.coe_eq_coe,
-    with_bot.coe_eq_coe, add_eq_zero_iff' (nat.zero_le _) (nat.zero_le _)]
-
-lemma with_bot.add_eq_one_iff : ∀ {n m : with_bot ℕ}, n + m = 1 ↔ (n = 0 ∧ m = 1) ∨ (n = 1 ∧ m = 0)
-| none     none     := dec_trivial
-| none     (some m) := dec_trivial
-| (some n) none     := iff_of_false dec_trivial (λ h, h.elim (λ h, absurd h.2 dec_trivial)
-  (λ h, absurd h.2 dec_trivial))
-| (some n) (some 0) := by erw [with_bot.coe_eq_coe, with_bot.coe_eq_coe, with_bot.coe_eq_coe,
-    with_bot.coe_eq_coe]; simp
-| (some n) (some (m + 1)) := by erw [with_bot.coe_eq_coe, with_bot.coe_eq_coe, with_bot.coe_eq_coe,
-    with_bot.coe_eq_coe, with_bot.coe_eq_coe]; simp [nat.add_succ, nat.succ_inj', nat.succ_ne_zero]
-
-@[simp] lemma with_bot.coe_nonneg {n : ℕ} : 0 ≤ (n : with_bot ℕ) :=
-by rw [← with_bot.coe_zero, with_bot.coe_le_coe]; exact nat.zero_le _
-
-@[simp] lemma with_bot.lt_zero_iff (n : with_bot ℕ) : n < 0 ↔ n = ⊥ :=
-option.cases_on n dec_trivial (λ n, iff_of_false
-  (by simp [with_bot.some_eq_coe]) (λ h, option.no_confusion h))
-
-lemma with_bot.one_le_iff_zero_lt {x : with_bot ℕ} : 1 ≤ x ↔ 0 < x :=
+namespace with_bot
+
+lemma add_eq_zero_iff {n m : with_bot ℕ} : n + m = 0 ↔ n = 0 ∧ m = 0 :=
+begin
+  rcases ⟨n, m⟩ with ⟨_ | _, _ | _⟩,
+  any_goals { tautology },
+  repeat { erw [with_bot.coe_eq_coe] },
+  exact add_eq_zero_iff
+end
+
+lemma add_eq_one_iff {n m : with_bot ℕ} : n + m = 1 ↔ n = 0 ∧ m = 1 ∨ n = 1 ∧ m = 0 :=
+begin
+  rcases ⟨n, m⟩ with ⟨_ | _, _ | _⟩,
+  any_goals { tautology },
+  repeat { erw [with_bot.coe_eq_coe] },
+  exact add_eq_one_iff
+end
+
+lemma add_eq_two_iff {n m : with_bot ℕ} :
+  n + m = 2 ↔ n = 0 ∧ m = 2 ∨ n = 1 ∧ m = 1 ∨ n = 2 ∧ m = 0 :=
+begin
+  rcases ⟨n, m⟩ with ⟨_ | _, _ | _⟩,
+  any_goals { tautology },
+  repeat { erw [with_bot.coe_eq_coe] },
+  exact add_eq_two_iff
+end
+
+lemma add_eq_three_iff {n m : with_bot ℕ} :
+  n + m = 3 ↔ n = 0 ∧ m = 3 ∨ n = 1 ∧ m = 2 ∨ n = 2 ∧ m = 1 ∨ n = 3 ∧ m = 0 :=
+begin
+  rcases ⟨n, m⟩ with ⟨_ | _, _ | _⟩,
+  any_goals { tautology },
+  repeat { erw [with_bot.coe_eq_coe] },
+  exact add_eq_three_iff
+end
+
+@[simp] lemma coe_nonneg {n : ℕ} : 0 ≤ (n : with_bot ℕ) :=
```

Comment:
This is [nat.with_bot.coe_nonneg](https://leanprover-community.github.io/mathlib_docs/data/nat/with_bot.html#nat.with_bot.coe_nonneg).